### DISCUSSION TITLE
How to get changed files before triggering workflow_call ?

### DISCUSSION THREAD

**Discussion by rffontenelle** [community/community#38015] [Actions] [answered]
I would like to set a GitHub Actions workflow to call another workflow in the same repository only when specified files get changed. I was planning to use [Reusing workflows](https://docs.github.com/en/actions/using-workflows/reusing-workflows) instructions but I don't know how I would get the list of changed files to run a "if" condition check.

So, how to get changed files before triggering workflow_call ?

Explaining my scenario:
- An workflow with a `on.workflow_call` event trigger configured, and it should receive 'language' input variable and build the docs contained in my repo
- Another workflow with a `on.push` and `on.pull_request` event trigger configured, and it should call the above workflow passing the language code (pt_BR, fr, de, etc.) as input. However, that call should only happen if the commit contains files in that language directory (e.g. call build of pt_BR only if commit has changes in pt_BR/ directory)

**Comment by airtower-luna** [comment, marked as answer]
> So, how to get changed files before triggering workflow_call ?

By adding another job that does the required processing, and creates a job output that says which language(s) you want to build for. Then you can have the job that calls the reusable workflow use that output to decide if it should run, and for which languages.

If the reusable workflow is intended to run for exactly one language a [matrix strategy](https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-a-matrix-strategy-with-a-reusable-workflow) might be useful to call it separately for all required languages.

**Reply by rffontenelle** [reply]
Following your suggestion I searched and found [Defining outputs for jobs](https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs) which made it clear how I should implement it. Thanks.
